lib/vscode/src/vs/server/node/server.ts

```diff
@@ -286,7 +302,15 @@ export class Vscode {
 				));
 				this.ipc.registerChannel('request', new RequestChannel(accessor.get(IRequestService)));
 				this.ipc.registerChannel('telemetry', new TelemetryChannel(telemetryService));
-				this.ipc.registerChannel('localizations', <IServerChannel<any>>createChannelReceiver(accessor.get(ILocalizationsService)));
+				/*
+					NOTE@coder: they renamed createChannelReceiver and made it part of the ProxyChannel namespace
```

Comment:
Totally see where you're coming from. 

My thought is if these messages are right next to the code, it's much easier/faster for a future engineer to understand context around why a change was made vs. trying to look through commit history + messages.

I'm new to writing longer commit messages so happy to be convinced otherwise!

Reply:
In my mind one of the main problems with maintaining a changelog next to the changed code is that the code no longer exists.

So for a future engineer `renamed createChannelReceiver` has no meaning since they've never seen a `createChannelReceiver` and probably never will, unless they go to the commit history to find it which sounds like it would defeat the purpose of the comment.

Or in other words, I think the question here to ask would be whether this comment enhances a future engineer's abililty to understand and maintain the code as it currently stands and if so how.

Actually I had to go to the diff anyway to understand what changed and I wrote the original hahaha. I couldn't remember what that function was or did, how or where it was used differently, if the new method was truly equivalent, etc. 

For me the second biggest problem is that there are already really good tools purpose-built for describing changes to code; the commenting system is almost a sort of a poor man's git.

VS Code for example with the git extension (I forget if it's the built-in or the GitLens one) can show you commit messages right in the file. This lets you see the reasoning for every line. It should also be possible to do things like traverse backwards to see how the code has evolved or browse a specific commit so you could easily look up what exactly `createChannelReceiver` was, for example.

I hope that's persuasive; if I was a video game character I'd fail every speech check.